cmd/update.go

```diff
@@ -0,0 +1,197 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/spf13/cobra"
+
+	"github.com/anchore/grype/internal/log"
+	"github.com/anchore/grype/internal/version"
+)
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "updates grype",
+	RunE:  runUpdate,
+}
+
+func init() {
+	rootCmd.AddCommand(updateCmd)
+}
+
+func runUpdate(_ *cobra.Command, _ []string) error {
+	fmt.Println("Checking for a newer version...")
+	newerVersionAvailable, desiredVersion, err := checkLatestVersion()
+	if err != nil {
+		panic(err)
+	}
+	if newerVersionAvailable {
+		compressed := downloadCompressed(desiredVersion)
+		filePath := extractBinary(compressed)
+		replaceBinary(filePath)
+	} else {
```

Comment:
consider doing a return after `replaceBinary` - this removes the extra `else` block and the default case for this function can be informing the user the newest version is already installed